assets/app/scripts/directives/truncate.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  // Truncates text to a length, adding a tooltip and an ellipsis if truncated.
+  // Different than `text-overflow: ellipsis` because it allows for multiline text.
+  .directive('truncateLongText', function() {
+    return {
+      restrict: 'E',
+      scope: {
+        content: '=',
+        limit: '=',
+        useWordBoundary: '='
+      },
+      template: '<span ng-attr-title="{{content}}">{{visibleContent}}<span ng-if="truncated">&hellip;</span></span>',
+      link: function(scope, elem, attr) {
+        scope.visibleContent = scope.content;
+        if (!scope.limit) {
```

Comment:
take this out... still need to watch to update visibleContent, and realistically, using this directive without a limit is silly